wordpress-mobile/WordPress-Android
# Changing lifecycle owner to be the fragment instead of the activity.

Fixes #10948 

This issue is related to the zalpha-199 and zalpha-200 versions.

We were able to reproduce the issue (see below for the steps). Not fully sure on the why, but what happens is that the observer lifecycle owner in the fragment is the getActivity() context and it can happen that the observer stays active even with a not valid activity to which currently the fragment is attached. We changed the lifecycle owner to be the Fragment instead of the Activity (that probably is also more correct).

## To reproduce (before applying the current PR)
- In an emulator (or a usb connected device to check logcat logs) open the reader so it is the last screen used in the app
- Close the app
- Force a notification for example making a new post in a site you follow
- Tap on the notification to open the new post
- The app does not open and in logcat you can see the following

```
NullPointerException: Attempt to invoke virtual method 'android.content.Context android.content.Context.getApplicationContext()' on a null object reference
    at org.wordpress.android.ui.reader.adapters.ReaderPostAdapter.<init>(ReaderPostAdapter.java:697)
    at org.wordpress.android.ui.reader.ReaderPostListFragment.getPostAdapter(ReaderPostListFragment.java:1809)
    at org.wordpress.android.ui.reader.ReaderPostListFragment.resetPostAdapter(ReaderPostListFragment.java:628)
    at org.wordpress.android.ui.reader.ReaderPostListFragment.changeReaderMode(ReaderPostListFragment.java:476)
    at org.wordpress.android.ui.reader.ReaderPostListFragment.lambda$onActivityCreated$2$ReaderPostListFragment(ReaderPostListFragment.java:411)
...
```
## To test
- Follow the above step and check the post opens in the app and no crash log in logcat
- smoke test the reader


PR submission checklist:
